### PR TITLE
add corner axis back and scatter selection on toolbar

### PR DESCRIPTION
**Pull request by PennyQ**
While the subset implement code within `on_mouse_release` doesn't work well, only one point will be highlighted instead of a real selection region. 

And the ellipse icon and point icon should be added into the toolbar.

### CODE REVIEW

File: glue_vispy_viewers/common/toolbar.py [hunk 1]
```diff
@@ -125,31 +172,163 @@ def on_mouse_release(self, event):
...
+        if event.button == 1 and self.mode is not None and self.mode is not 'point':
+            # self.facecolor[self.facecolor[:, 1] != 1.0] = self.white
+            data = tr.map(layer_data)[:, :2]
+
+            selection_path = path.Path(self.line_pos, closed=True)
+
+            mask = [selection_path.contains_points(data)]
+
+            self.selected = mask
+            self.mark_selected()
```
Comment (PennyQ) [inline glue_vispy_viewers/common/toolbar.py]: I use this `mark_selected()` for highlighting to show that the mask is corrected extracted, it should be replaced with the `Subset` operation later.

File: glue_vispy_viewers/common/toolbar.py [hunk 2]
```diff
@@ -125,31 +172,163 @@ def on_mouse_release(self, event):
...
+            # subset_state = ElementSubsetState(np.where(mask)[0])
 
-        # TODO: here we need to get a mask for the selection, which we assume
-        # will be called ``mask``. For now, we set the mask to a random mask.
-        mask = np.random.random(visible_data[0].shape) > 0.5
+            # We now check what the selection mode is, and update the selection as
+            # needed (this is delegated to the correct subset mode).
+            # mode = EditSubsetMode()
+            # focus = visible_data[0] if len(visible_data) > 0 else None
+            # mode.update(self._data_collection, subset_state, focus_data=focus)
```
Comment (PennyQ) [inline glue_vispy_viewers/common/toolbar.py]: I use the `mark_selected()` for highlighting to show that the mask is corrected extracted, it should be replaced with the `Subset` operation later.

**Comment by astrofrog** [comment]
@PennyQ - thanks! Is the point selection icon not already in the toolbar? You can add an ellipse icon similarly to the other icons (e.g. `glue_lasso`) and using `glue_circle` for the icon name.

**Comment by PennyQ** [comment]
@astrofrog No, the points selection doesn't show up, I added the ellipse selection here and now trying to implement the volume selection.

The checks failed because of the `freeze()` in Vispy, which is used by the corner axis within this PR. But it works well with my old Vispy (0.4.0.106.gf8a69d0), so strange :(

**Comment by PennyQ** [comment]
I post a new commit in which the `toolbar` class is divided to a base one and scatter/volume ones, so we could implement different selection methods for scatter/volume data later. The scatter toolbar works well now, except the toolbar is drawn twice now...
While I found `data getting & setting` from MultipleVolume visual is very complex, could you show me how to do it or I could write a `frame code` within this PR and leave it for you @astrofrog ?
Thanks =)

**Comment by astrofrog** [comment]
@PennyQ - are you or @maxwelltsai familiar with cherry-picking and rebasing? If so, could you split out the changes for the 3D axis into another pull request? (it's better to keep one idea per pull request, otherwise we won't be able to merge the selection changes until the tests pass for the axis).

As I mentioned before, you should not open pul requests from the master branch, but instead make one branch per feature. The easiest might be to create two new branches, copy the commits you want into each one, and then open a pull request from both branches. If you are not familiar with this, I can take you through the steps tomorrow when we speak?

**Comment by PennyQ** [comment]
@astrofrog Yeah I used `rebase` before, I got your point and I will close this one first and create two new PR for each of new features. Thanks for showing me this :)

**Pull request closed without merging**